parser/encoder.go

```diff
@@ -79,8 +79,13 @@ func (e *encoder) encodeAsString(packet *Packet) types.BufferInterface {
 	}
 	// json data
 	if nil != packet.Data {
-		if b, err := json.Marshal(_encodeData(packet.Data)); err == nil {
-			str.Write(b)
+		if pds, is := packet.Data.(string); is {
```

Comment:
This behavior seems inconsistent with nodejs socket.io-parser, which will cause unpredictable problems in other clients:
test golang:
```golang
package main

import (
	"fmt"

	"github.com/zishang520/socket.io-go-parser/v2/parser"
)

func main() {
	en := parser.NewEncoder()

	id := uint64(1)
	Attachments := uint64(1)

	fmt.Println(en.Encode(&parser.Packet{
		Type:        parser.EVENT,
		Nsp:         "/a",
		Data:        "any",
		Id:          &id,
		Attachments: &Attachments,
	}))
}
```
output: [2/a,1any]

test nodejs (latest):
```nodejs
const {
    Encoder,
    PacketType
} = require("socket.io-parser");
let en = new Encoder;

console.log(en.encode({
    type: PacketType.EVENT,
    nsp: "/a",
    data: "any",
    id: 1,
    attachments: 1,
}))
```

output:  [ '2/a,1"any"' ]

Reply:
Ah, wasn't aware you could use it that way.  Adding a private field to store it instad...